### PR TITLE
Add first class svg mode.

### PR DESCRIPTION
**Pull request by Raynos**
This allows people to generate an svg file in one step.

The approach taken is to copy the perl scripts in and
spawn them as child processes.

r: @jcorbin @danielheller @Corgiman

### CODE REVIEW

File: cmd/torch.js [hunk 1]
```diff
@@ -134,6 +135,79 @@ function outputDTraceText(stacks) {
...
+            'bg-flamegraphs',
+            'stackcollapse-stap.pl'
+        );
+
+        var command = [
+            'perl ',
+            stackCollapse,
+            ' ',
+            tempFile,
+            ' | tr -d "\\0"'
```
Comment (jcorbin) [inline cmd/torch.js]: at least add a note that this is a hack around broken output from this same project, and should be removed once said bug is fixed.

File: cmd/torch.js [hunk 2]
```diff
@@ -134,6 +135,79 @@ function outputDTraceText(stacks) {
...
+    var tempFile = path.join(os.tmpDir(), 'stack.raw');
+    var tempFile2 = path.join(os.tmpDir(), 'stack.folded');
+
+    adaptor.pipe(fs.createWriteStream(tempFile))
+        .once('finish', fileWritten);
+
+    adaptor.resume();
+
+    function fileWritten() {
+        var stackCollapse = path.join(
```
Comment (jcorbin) [inline cmd/torch.js]: hoist the path resolutions to module level

**Comment by sandlerben** [comment]
When I tried to add the flamegraph.pl script to the [go-torch](https://github.com/uber/go-torch) repo, legal said it was not okay because of its CDDL license.

**Comment by jcorbin** [comment]
Okay so can we just get the npm install to download it as dep into
node_modules?

**Comment by Raynos** [comment]
I'll update the code to fetch them into /tmp from github at run time.

**Comment by Raynos** [comment]
RIP

**Pull request closed without merging**